### PR TITLE
dynamic snap writelocking, node deletion plumbing, storm priv elevation, and *type= lift cmpr

### PR DESCRIPTION
**Pull request by invisig0th**

### CODE REVIEW

File: synapse/lib/layer.py [hunk 1]
```diff
@@ -73,10 +98,17 @@ def abort(self):
         self.xact.abort()
 
     def commit(self):
+
         # committing on a write transaction on a different thread than it was created is fatal
         assert(not self.write or self.tid == s_threads.iden())
```
Comment (vertexmc) [inline synapse/lib/layer.py]: Would rather check and raise

File: synapse/lib/node.py [hunk 2]
```diff
@@ -382,4 +356,64 @@ def delTag(self, tag, init=False):
...
+        tags = [(len(t), t) for t in self.tags.keys()]
+
+        # check for tag permissions
+        for size, tag in tags:
+            tagpath = s_chop.tagpath(tag)
+            if not self.snap.allowed('tag:del', *tagpath):
+                return self.snap._onAuthDeny('Not allowed to delete node with tag {tag}.')
+
+        # check for any nodes which reference us...
+        if any(self.snap._getNodesByType(formname, formvalu, addform=False)):
```
Comment (vertexmc) [inline synapse/lib/node.py]: We should include unit test to ensure we can't end up in situation where two nodes that refer to each other can't be deleted

File: synapse/lib/storm.py [hunk 3]
```diff
@@ -101,13 +101,52 @@ def getArgParser(self):
...
+    def runStormCmd(self, snap, genr):
+
+        # a bit odd, but we need to be detected as a generator
+        yield from ()
+
+        if self.opts.sudo:
+            snap.elevated = True
+
+        for node in genr:
+            node.delete()
```
Comment (vEpiphyte) [inline synapse/lib/storm.py]: Do we want to yield a message indicating the success of deleting a node?

File: synapse/models/infotech.py [hunk 4]
```diff
@@ -310,13 +310,16 @@ def getModelDefs(self):
...
                     }),
                     ('author:org', ('ou:org', {}), {
-                        'doc': 'Organization responsible for the software.',
+                        'doc': 'Organization which authored the software',
                     }),
                     ('author:acct', ('inet:web:acct', {}), {
-                        'doc': 'Web user responsible for the software.',
+                        'doc': 'Web account of the software author.',
+                    }),
+                    ('author:email', ('inet:email', {}), {
```
Comment (vEpiphyte) [inline synapse/models/infotech.py]: software test needs to be updated to set this property

File: synapse/lib/storm.py (hunk 3 above)
Comment (vertexmc) [reply synapse/lib/storm.py]: It would be nice to know which nodes were deleted

Comment (invisig0th) [reply synapse/lib/storm.py]: The node:del splice should fulfill this purpose.

Comment (vertexmc) [reply synapse/lib/storm.py]: Maybe even a message like "N nodes deleted"

**Pull request merged**